### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by Emma-Naumann**
@Liam-Eime please review this work by checking for best practice use of functions, naming and code patterns, and ensure all questions have been answered properly etc..

### CODE REVIEW

File: prac_05/wimbledon.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    display_results(champion_to_count, countries)
+
+
+def get_records(filename):
+    """Get records from file and store as a list of lists."""
+    records = []
+    with open(FILENAME, "r", encoding="utf-8-sig") as in_file:
+        in_file.readline()
+        for line in in_file:
+            parts = line.strip().split(",")
```
Comment (Liam-Eime) [inline prac_05/wimbledon.py:26]: I don't know if .strip() is necessary for the desired outcome to work, I would recommend giving it a go without it :)

File: prac_05/wimbledon.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return champion_to_count, countries
+
+
+def display_results(champion_to_count, countries):
+    """Display champions and countries."""
+    print("wimbledon Champions:")
+    for champion, count in champion_to_count.items():
+        print(f"{champion} {count}")
+    print(f"These {len(countries)} countries have won Wimbledon:")
+    print(", ".join(country for country in sorted(countries)))
```
Comment (Liam-Eime) [inline prac_05/wimbledon.py]: You might be able to just use .join on the sorted set of countries, maybe give it a go for yourself to see?

File: prac_05/wimbledon.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return records
+
+
+def process_records(records):
+    """Create dictionary of champions and set of countries from records."""
+    champion_to_count = {}
+    countries = set()
+    for record in records:
+        countries.add(record[INDEX_COUNTRY])
+        try:
```
Comment (Liam-Eime) [inline prac_05/wimbledon.py:37]: Just a note that there are two alternatives way to do this if you wanted to. Both are in the lecture material slides and explained in the videos. The third way of the three total ways is shorter if you wanted to explore that option :)

File: prac_05/wimbledon.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+"""
+Emma Naumann
+CP1404 Practical 5
+Estimate: 30 minutes
+Actual:
+"""
+
+FILENAME = "wimbledon.csv"
+INDEX_COUNTRY = 1
```
Comment (Liam-Eime) [inline prac_05/wimbledon.py]: Index of country? You could name it as COUNTRY_INDEX to say that it is the index in which the country is located. Might be something to think about :)

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+INDEX_CHAMPION = 2
```
Comment (Liam-Eime) [inline prac_05/wimbledon.py]: Same as the country index :)

File: prac_05/word_occurences.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+"""
+Emma Naumann
+CP1404 Practical 5
+
```
Comment (Liam-Eime) [inline prac_05/word_occurences.py]: What does this program do? Doc strings are nice when they say what the program does and is a helpful reminder to yourself when programming as to what your goal is

File: prac_05/state_names.py [hunk 7]
```diff
@@ -5,15 +5,22 @@
...
+    try:
         print(state_code, "is", CODE_TO_NAME[state_code])
-    else:
-        print("Invalid short state")
-    state_code = input("Enter short state: ")
+    except KeyError:
+        print("Invalid short state. Try again.")
+    # if state_code in CODE_TO_NAME:
+    #     print(state_code, "is", CODE_TO_NAME[state_code])
+    # else:
```
Comment (Liam-Eime) [inline prac_05/state_names.py:21]: Can you remove the commented out if statement?

File: prac_05/wimbledon.py (hunk 2 above)
Comment (Emma-Naumann) [reply prac_05/wimbledon.py]: Good point. Thanks.

File: prac_05/wimbledon.py (hunk 4 above)
Comment (Emma-Naumann) [reply prac_05/wimbledon.py]: Fixed. Thanks :)

File: prac_05/word_occurences.py (hunk 6 above)
Comment (Emma-Naumann) [reply prac_05/word_occurences.py]: 👍

**Comment by Emma-Naumann** [comment]
Thanks @Liam-Eime for the review :)

**Pull request merged**